### PR TITLE
WPF-60935 : Update the UG for Asynchronous Load-On-Demand support

### PR DESCRIPTION
**Pull request by VasanthkumarDurairaj**
I have update the UG for Asynchronous Load-On-Demand support

### CODE REVIEW

**Review by susmitha-sundar — APPROVED** [review]

File: wpf/TreeGrid/Load-On-Demand.md [hunk 1]
```diff
@@ -409,5 +409,162 @@ You can let SfTreeGrid to populate the data at runtime by calling [SfTreeGrid.Re
 
 N> [View Sample in GitHub](https://github.com/SyncfusionExamples/How-to-load-data-on-demand-using-events-in-wpf-treegrid).
 
+## Asynchronous load on demand using event
 
-N> You can refer to our [WPF TreeGrid](https://www.syncfusion.com/wpf-controls/treegrid) feature tour page for its groundbreaking feature representations. You can also explore our [WPF TreeGrid example](https://github.com/syncfusion/wpf-demos) to know how to render and configure the treegrid.
\ No newline at end of file
+The SfTreeGrid provides support for loading data asynchronously on-demand by using the [SfTreeGrid.RequestTreeItems](https://help.syncfusion.com/cr/wpf/Syncfusion.UI.Xaml.TreeGrid.SfTreeGrid.html#Syncfusion_UI_Xaml_TreeGrid_SfTreeGrid_RequestTreeItems) event. This event enables you to fetch both parent and child items asynchronously.
```
Comment (Chrispineagunja) [inline wpf/TreeGrid/Load-On-Demand.md:414]: The SfTreeGrid supports loading data asynchronously on-demand by using the [SfTreeGrid.RequestTreeItems](https://help.syncfusion.com/cr/wpf/Syncfusion.UI.Xaml.TreeGrid.SfTreeGrid.html#Syncfusion_UI_Xaml_TreeGrid_SfTreeGrid_RequestTreeItems) event. This event enables you to fetch both parent and child items asynchronously.

File: wpf/TreeGrid/Load-On-Demand.md [hunk 2]
```diff
@@ -409,5 +409,162 @@ You can let SfTreeGrid to populate the data at runtime by calling [SfTreeGrid.Re
 
 N> [View Sample in GitHub](https://github.com/SyncfusionExamples/How-to-load-data-on-demand-using-events-in-wpf-treegrid).
 
+## Asynchronous load on demand using event
 
-N> You can refer to our [WPF TreeGrid](https://www.syncfusion.com/wpf-controls/treegrid) feature tour page for its groundbreaking feature representations. You can also explore our [WPF TreeGrid example](https://github.com/syncfusion/wpf-demos) to know how to render and configure the treegrid.
\ No newline at end of file
+The SfTreeGrid provides support for loading data asynchronously on-demand by using the [SfTreeGrid.RequestTreeItems](https://help.syncfusion.com/cr/wpf/Syncfusion.UI.Xaml.TreeGrid.SfTreeGrid.html#Syncfusion_UI_Xaml_TreeGrid_SfTreeGrid_RequestTreeItems) event. This event enables you to fetch both parent and child items asynchronously. 
+
+In the below example, the SfTreeGrid retrieves parent and child nodes asynchronously by using the `RequestTreeItems` event.
```
Comment (Chrispineagunja) [inline wpf/TreeGrid/Load-On-Demand.md:416]: In the following example, the SfTreeGrid retrieves parent and child nodes asynchronously by using the `RequestTreeItems` event.

File: wpf/TreeGrid/Load-On-Demand.md [hunk 3]
```diff
@@ -409,5 +409,162 @@ You can let SfTreeGrid to populate the data at runtime by calling [SfTreeGrid.Re
...
+treeGrid.RequestTreeItems += TreeGrid_RequestTreeItems;
+      
+private async void TreeGrid_RequestTreeItems(object sender, TreeGridRequestTreeItemsEventArgs args)
+{
+
+    if (args.ParentItem == null)
+    {
+        var parentItems = viewModel.EmployeeDetails.Where(x => x.ReportsTo == -1);
+
+        // time delay to retrieve the parent items.
```
Comment (Chrispineagunja) [inline wpf/TreeGrid/Load-On-Demand.md:430]: // Time delay to retrieve the parent items.

File: wpf/TreeGrid/Load-On-Demand.md [hunk 4]
```diff
@@ -409,5 +409,162 @@ You can let SfTreeGrid to populate the data at runtime by calling [SfTreeGrid.Re
...
+
+    else
+    {
+        EmployeeInfo employee = args.ParentItem as EmployeeInfo;
+
+        if (employee != null)
+        {
+            var childItems = viewModel.GetEmployees().Where(x => x.ReportsTo == employee.ID);
+
+            // time delay to retrieve the child items.
```
Comment (Chrispineagunja) [inline wpf/TreeGrid/Load-On-Demand.md:444]: // Time delay to retrieve the child items.

File: wpf/TreeGrid/Load-On-Demand.md [hunk 5]
```diff
@@ -409,5 +409,162 @@ You can let SfTreeGrid to populate the data at runtime by calling [SfTreeGrid.Re
...
+
+        return employeeDetails;
+    }
+}
+{% endhighlight %}
+{% endtabs %}
+
+![WPF TreeGrid Asynchronous Load on demand support](Getting-Started_images/wpf-treegrid-async-on-demand-load.gif)
+
+N> You can refer to our [WPF TreeGrid](https://www.syncfusion.com/wpf-controls/treegrid) feature tour page for its groundbreaking feature representations. You can also explore our [WPF TreeGrid example](https://github.com/syncfusion/wpf-demos) to know how to render and configure the treegrid.
```
Comment (Chrispineagunja) [inline wpf/TreeGrid/Load-On-Demand.md:570]: N> Refer to our [WPF TreeGrid](https://www.syncfusion.com/wpf-controls/treegrid) feature tour page for its groundbreaking feature representations. You can also explore our [WPF TreeGrid example](https://github.com/syncfusion/wpf-demos) to learn how to render and configure the treegrid.

Comment (VasanthkumarDurairaj) [reply wpf/TreeGrid/Load-On-Demand.md:570]: Updated the changes

File: wpf/TreeGrid/Load-On-Demand.md (hunk 4 above)
Comment (VasanthkumarDurairaj) [reply wpf/TreeGrid/Load-On-Demand.md:444]: Updated the changes

File: wpf/TreeGrid/Load-On-Demand.md (hunk 3 above)
Comment (VasanthkumarDurairaj) [reply wpf/TreeGrid/Load-On-Demand.md:430]: Updated the changes

File: wpf/TreeGrid/Load-On-Demand.md (hunk 2 above)
Comment (VasanthkumarDurairaj) [reply wpf/TreeGrid/Load-On-Demand.md:416]: Updated the changes

File: wpf/TreeGrid/Load-On-Demand.md (hunk 1 above)
Comment (VasanthkumarDurairaj) [reply wpf/TreeGrid/Load-On-Demand.md:414]: Updated the changes

**Review by susmitha-sundar — APPROVED** [review]

**Pull request merged**